Lib/importlib/metadata/_adapters.py

```diff
@@ -87,3 +181,33 @@ def transform(key):
             return tk, value
 
         return dict(map(transform, map(FoldedCase, self)))
+
+    def _parse_idents(self, s):
+        es = (i[0] for i in _entries.findall(s))
+        es = (_name_email.match(i).groups()[::3] for i in es)
```

Comment:
I recommend to avoid overloading the meaning of a variable. It forces the reader to carry the variable contexts while reading through the logic and makes debugging harder (when prior values are no longer reachable). I suggest to think of a different variable name for the various stages of progression, something like `found`, `matched`, ...

Reply:
Hmm, not sure I agree. More names means more work and more connections for the reader to make. Which do you prefer?

```
def _parse_idents(self, string):
    entries = (entry[0] for entry in _entries.findall(string))
    entries = (_name_email.match(entry).groups()[::3] for entry in entries)
    entries = {Ident(*entry) for entry in entries if entry != (None, None)}
    return entries
```

```
def _parse_idents(self, string):
    entries = (entry[0] for entry in _entries.findall(string))
    cleaved = (_name_email.match(entry).groups()[::3] for entry in entries)
    updated = {Ident(*entry) for entry in cleaved if entry != (None, None)}
    return updated
```

Reply:
In the end, it doesn't matter.